plugins/view-resources/src/components/EditDoc.svelte

```diff
@@ -0,0 +1,328 @@
+    )
+    return mixins.map((m) => hierarchy.getClass(m) as Mixin<Doc>)
+  }
+
+  function filterKeys (keys: KeyedAttribute[], ignoreKeys: string[]): KeyedAttribute[] {
+    keys = keys.filter((k) => !docKeys.has(k.key))
+    keys = keys.filter((k) => !ignoreKeys.includes(k.key))
+    return keys
+  }
+
+  function getFiltredKeys (objectClass: Ref<Class<Doc>>, ignoreKeys: string[], to?: Ref<Class<Doc>>): KeyedAttribute[] {
+    const keys = [...hierarchy.getAllAttributes(objectClass, to).entries()]
+      .filter(([, value]) => value.hidden !== true)
+      .map(([key, attr]) => ({ key, attr }))
+
+    return filterKeys(keys, ignoreKeys)
+  }
+
+  function updateKeys (ignoreKeys: string[]): void {
+    const filtredKeys = getFiltredKeys(
+      selectedClass ?? object._class,
+      ignoreKeys,
+      selectedClass !== objectClass._id ? objectClass._id : undefined
+    )
+    keys = collectionsFilter(filtredKeys, false)
+    collectionKeys = collectionsFilter(filtredKeys, true)
+  }
+
+  function collectionsFilter (keys: KeyedAttribute[], get: boolean): KeyedAttribute[] {
+    const result: KeyedAttribute[] = []
+    for (const key of keys) {
+      if (isCollectionAttr(key) === get) result.push(key)
+    }
+    return result
+  }
+
+  function isCollectionAttr (key: KeyedAttribute): boolean {
+    return hierarchy.isDerived(key.attr.type._class, core.class.Collection)
+  }
+
+  async function getEditor (_class: Ref<Class<Doc>>): Promise<AnyComponent> {
+    const clazz = hierarchy.getClass(_class)
+    const editorMixin = hierarchy.as(clazz, view.mixin.ObjectEditor)
+    if (editorMixin?.editor == null && clazz.extends != null) return getEditor(clazz.extends)
+    return editorMixin.editor
+  }
+
+  async function getEditorOrDefault (
+    _class: Ref<Class<Doc>> | undefined,
+    defaultClass: Ref<Class<Doc>>
+  ): Promise<AnyComponent> {
+    const editor = _class !== undefined ? await getEditor(_class) : undefined
+    if (editor !== undefined) {
+      return editor
+    }
+    return getEditor(defaultClass)
+  }
+
+  async function getCollectionEditor (key: KeyedAttribute): Promise<AnyComponent> {
+    const attrClass = getAttributePresenterClass(key.attr)
+    const clazz = client.getHierarchy().getClass(attrClass)
+    const editorMixin = client.getHierarchy().as(clazz, view.mixin.AttributeEditor)
+    return editorMixin.editor
+  }
+
+  function getIcon (_class: Ref<Class<Obj>>): Asset {
+    let clazz = hierarchy.getClass(_class)
+    if (clazz.icon !== undefined) return clazz.icon
+    while (clazz.extends !== undefined) {
+      clazz = hierarchy.getClass(clazz.extends)
+      if (clazz.icon !== undefined) {
+        return clazz.icon
+      }
+    }
+    throw new Error(`Icon not found for ${_class}`)
+  }
+
+  $: icon = object && getIcon(object._class)
+
+  let mainEditor: HTMLElement
+  let prevEditor: HTMLElement
+  let maxHeight = 0
+  const observer = new ResizeObserver(() => {
```

Comment:
We discussed with @aplatoff about same size headers for all mixins per/class. So it is ok to have, but it should not be necessary.